atom/renderer/api/atom_api_web_frame.cc

```diff
@@ -368,6 +374,102 @@ void WebFrame::ClearCache(v8::Isolate* isolate) {
     base::MemoryPressureListener::MEMORY_PRESSURE_LEVEL_CRITICAL);
 }
 
+v8::Local<v8::Value> WebFrame::Opener() const {
+  blink::WebLocalFrame* local_frame =
+    WebFrame::web_frame_->Opener()->ToWebLocalFrame();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::Parent() const {
+  blink::WebLocalFrame* local_frame =
+    WebFrame::web_frame_->Parent()->ToWebLocalFrame();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::Top() const {
+  blink::WebLocalFrame* local_frame =
+    WebFrame::web_frame_->Top()->ToWebLocalFrame();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::FirstChild() const {
+  blink::WebLocalFrame* local_frame =
+    WebFrame::web_frame_->FirstChild()->ToWebLocalFrame();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::NextSibling() const {
+  blink::WebLocalFrame* local_frame =
+    WebFrame::web_frame_->NextSibling()->ToWebLocalFrame();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::TraverseNext() const {
+  blink::WebLocalFrame* local_frame =
+    WebFrame::web_frame_->TraverseNext()->ToWebLocalFrame();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::GetFrameForSelector(mate::Arguments* args)
```

Comment:
You can define the function as `GetFrameForSelector(const std::string& selector)` and the argument would be converted automatically. 

This also applies to `FindFrameByName`.